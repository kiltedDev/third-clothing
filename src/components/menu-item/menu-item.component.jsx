import React from 'react';
import { withRouter } from 'react-router-dom';

import {
  MenuItemContainer,
  BackgroundImageContainer,
  ContentContainer,
  ContentTitle,
  ContentSubtitle
} from './menu-item.styles';

const MenuItem = ({ title, imageURL, size, history, linkURL, match }) => (
  <MenuItemContainer
    className='menu-item'
    size={ size }
    onClick={ () =>
      history.push(`${match.url}${linkURL}`)
    }
  >
    <BackgroundImageContainer
      className='background-image'
      imageUrl={ imageURL }
    />
    <ContentContainer className='content'>
      <ContentTitle
        className='title'
        children={ title.toUpperCase() }
      />
      <ContentSubtitle
        className='subtitle'
        children='SHOP NOW'
      />
    </ContentContainer>
  </MenuItemContainer>
)

export default withRouter( MenuItem );
