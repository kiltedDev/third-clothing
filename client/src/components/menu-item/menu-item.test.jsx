import { shallow } from 'enzyme';
import React from 'react';
import { MenuItem } from './menu-item.component';

describe('MenuItem component', () => {
  let wrapper;
  let snapshot;
  let mockMatch;
  let mockHistory;
  const linkURL = '/hats';
  const imageURL = 'www.coolimage.com';
  const size = 'large';

  beforeAll(() => {
    mockMatch = {
      url: '/shop'
    };
    mockHistory = {
      push: jest.fn()
    };
    const mockProps = {
      title: 'hats',
      imageURL: imageURL,
      size: size,
      history: mockHistory,
      linkURL: linkURL,
      match: mockMatch
    }
    wrapper = shallow(<MenuItem { ...mockProps } />);
    snapshot = wrapper.debug();
  })

  it('should render MenuItem component',() => {
    expect( snapshot ).toMatchSnapshot();
  })

  it('should push history and call the right addy when clicked',() => {
    wrapper.find('MenuItemContainer').simulate('click');
    expect( mockHistory.push ).toHaveBeenCalledWith(`${mockMatch.url}${linkURL}`);
  })

  it('should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(size);
  });

  it('should pass imageUrl to BackgroundImageContainer as the prop imageUrl', () => {
    expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe( imageURL );
  });
})
