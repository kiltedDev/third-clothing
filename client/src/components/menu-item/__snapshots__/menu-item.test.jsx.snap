// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render MenuItem component 1`] = `
"<MenuItemContainer className=\\"menu-item\\" size=\\"large\\" onClick={[Function: onClick]}>
  <BackgroundImageContainer className=\\"background-image\\" imageUrl=\\"www.coolimage.com\\" />
  <styled.div className=\\"content\\">
    <styled.span className=\\"title\\">
      HATS
    </styled.span>
    <styled.span className=\\"subtitle\\">
      SHOP NOW
    </styled.span>
  </styled.div>
</MenuItemContainer>"
`;
